### PR TITLE
Function indexes

### LINKED ISSUE #6580: [Breakpoints] Breakpoint changes its place on page refresh
### STR
```javascript
a = {
    sameName: () => {
        return 1; /* After refresh breakpoint is now set on this line instead. */
    }
}
b = {
    sameName: () => {
        return 42; /* Set breakpoint here and refresh.*/
    }
}
```
1. Set breakpoint on line with 'return 42'
2. Refresh page

Actual results:

Breakpoint is now set on line with 'return 1'.

Expected results:

Breakpoint should stay where it was set.

This happens when two or more methods have the same name even when they are part of different objects. 
If I had to guess debugger stores breakpoint line numbers as an offset from the line where method is defined. Then on refresh it finds the first time this method's name appears and sets the breakpoint with that offset there disregarding that there could be more than one method in the code with such name.

### PR DESCRIPTION
**Pull request by bomsy**
Fixes #6580 

### Summary of Changes

* Add a `index` property to uniquely identify the function scope 

### Test Plan
- Load https://lumpy-plier.glitch.me/
- Open the debugger and create a breakpoint on line 11 return 42
- Refresh the page
- The breakpoint is now on line 6 -> BUG

### CODE REVIEW

File: src/utils/breakpoint/astBreakpointLocation.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export function getASTLocation(
   location: Location
 ): ASTLocation {
   if (source.isWasm || !symbols || symbols.loading) {
-    return { name: undefined, offset: location };
+    return { name: undefined, offset: location, index: undefined };
```
Comment (jasonLaster) [inline src/utils/breakpoint/astBreakpointLocation.js]: maybe we should default to 0?

Comment (bomsy) [reply src/utils/breakpoint/astBreakpointLocation.js]: sure

File: src/actions/tests/helpers/breakpoints.js [hunk 2]
```diff
@@ -19,7 +19,8 @@ export function mockPendingBreakpoint(overrides = {}) {
       name: undefined,
       offset: {
         line: line || 5
-      }
+      },
+      index: undefined
```
Comment (jasonLaster) [inline src/actions/tests/helpers/breakpoints.js]: should this be undefined or 0?

Comment (bomsy) [reply src/actions/tests/helpers/breakpoints.js]: oooppps missed this!

**Comment by jasonLaster** [comment]
Sorry for the delay in landing this. This is a super nice bug fix!

**Pull request merged**